bitnami/minio/templates/statefulset.yaml

```diff
@@ -0,0 +1,175 @@
+        {{- if .Values.image.debug }}
+        - name: BITNAMI_DEBUG
+          value: "true"
+        {{- end }}
+        - name: MINIO_DISTRIBUTED_MODE_ENABLED
+          value: "yes"
+        - name: MINIO_SKIP_CLIENT
+          value: "yes"
+        - name: MINIO_DISTRIBUTED_NODES
+          {{- $minioFullname := include "minio.fullname" . }}
+          {{- $minioHeadlessServiceName := printf "%s-%s" $minioFullname "headless" | trunc 63  }}
+          {{- $releaseNamespace := .Release.Namespace }}
+          value: {{range $i, $e := until $replicaCount }}{{ $minioFullname }}-{{ $e }}.{{ $minioHeadlessServiceName }}.{{ $releaseNamespace }}.svc.cluster.local,{{ end }}
+        - name: MINIO_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              {{- if .Values.existingSecret }}
+              name: {{ .Values.existingSecret }}
+              {{- else }}
+              name: {{ include "minio.fullname" . }}
+              {{- end }}
+              key: access-key
+        - name: MINIO_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              {{- if .Values.existingSecret }}
+              name: {{ .Values.existingSecret }}
+              {{- else }}
+              name: {{ include "minio.fullname" . }}
+              {{- end }}
+              key: secret-key
+        - name: MINIO_BROWSER
+          {{- if .Values.disableWebUI }}
+          value: "off"
+          {{- else }}
+          value: "on"
+          {{- end }}
+{{- if .Values.extraEnv }}
+{{ toYaml .Values.extraEnv | indent 8 }}
+{{- end }}
+        ports:
+        - name: minio
+          containerPort: 9000
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          httpGet:
+            path: /minio/health/live
+            port: minio
+        {{- end }}
+        {{- if .Values.readinessProbe.enabled }}
+        readinessProbe:
+          initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          tcpSocket:
+            port: minio
+        {{- end }}
+        {{- if .Values.resources }}
+        resources: {{ toYaml .Values.resources | nindent 10 }}
+        {{- end }}
+        volumeMounts:
+        - name: "data"
+          mountPath: {{ .Values.persistence.mountPath }}
+{{- if and .Values.persistence.enabled .Values.persistence.existingClaim }}
```

Comment:
This will not to work unless it is a ReadWriteMany PVC. I would leave the existingClaim only for the standalone mode.

Reply:
I agree. It makes no sense to use the same claim for more that one node in distributed mode.